src/mutations/creators.ts

```diff
@@ -53,24 +54,46 @@ export const createTypeAdditionMutation = (
 		) ||
 		isKnownGlobalBaseType(declaredType)
 	) {
-		return {
-			insertion: ` ${newTypeAlias}`,
-			range: {
-				begin: node.type.pos,
-				end: node.type.end,
-			},
-			type: "text-swap",
-		};
+		return textSwap(` ${newTypeAlias}`, node.type.pos, node.type.end);
+	}
+
+	const isAsyncFunction =
+		ts.isFunctionDeclaration(node) &&
+		node.modifiers?.some((modifier) => tsutils.isAsyncKeyword(modifier));
+
+	if (!isAsyncFunction) {
+		// Create a mutation insertion that adds the missing types in
+		return textInsert(` | ${newTypeAlias}`, node.type.end);
+	}
+
+	const typeChecker = request.services.languageService
+		.getProgram()
+		?.getTypeChecker();
+
+	const typesAsPromises = new Set(
+		[...missingTypes].map((type) => {
+			const stringified = typeChecker?.typeToString(
+				typeChecker.getBaseTypeOfLiteralType(type),
+			);
+			return stringified?.startsWith("Promise<")
```

Comment:
[Bug] If the user defines their own `Promise<` class/type, this will false positive. 

You'll want to use something like https://typescript-eslint.io/packages/type-utils#issymbolfromdefaultlibrary instead of a stringification check.